BD-3019 Add WhatsApp SQL table references (#7314)
braze-inc/braze-docs: _docs/_user_guide/engagement_tools/segments/sql_segments/sql_segments_tables.md

[USERS_MESSAGES_SMS_SHORTLINKCLICK_SHARED](#USERS_MESSAGES_SMS_SHORTLINKCLICK_SHARED) | When a user clicks a Braze shortened URL included in an SMS message
[USERS_MESSAGES_WEBHOOK_ABORT_SHARED](#USERS_MESSAGES_WEBHOOK_ABORT_SHARED) | An originally scheduled webhook message was aborted for some reason.
[USERS_MESSAGES_WEBHOOK_SEND_SHARED](#USERS_MESSAGES_WEBHOOK_SEND_SHARED) | When we send a webhook for a user

> [USERS_MESSAGES_WHATSAPP_ABORT_SHARED](#USERS_MESSAGES_WHATSAPP_ABORT_SHARED) | An originally scheduled WhatsApp message was aborted for some reason.

Reviewer: ```suggestion
[USERS_MESSAGES_WHATSAPP_ABORT_SHARED](#USERS_MESSAGES_WHATSAPP_ABORT_SHARED) | An originally scheduled WhatsApp message was aborted for some reason
```
Suggestion to remove the period to match the style for the other descriptions

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`app_api_id` | `null,`&nbsp;`string` | API ID of the app on which this event occurred
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

> `language` | `null,`&nbsp;`string` | [PII] language of the user

Reviewer: ```suggestion
`language` | `null,`&nbsp;`string` | [PII] Language of the user
```
Should all of the mentions of "[PII]" be "(PII)" with the first word after capitalized?

`app_api_id` | `null,`&nbsp;`string` | API ID of the app on which this event occurred
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`app_api_id` | `null,`&nbsp;`string` | API ID of the app on which this event occurred
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`card_api_id` | `null,`&nbsp;`string` | API ID of the card
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`app_api_id` | `null,`&nbsp;`string` | API ID of the app on which this event occurred
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`app_api_id` | `null,`&nbsp;`string` | API ID of the app on which this event occurred
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`app_api_id` | `null,`&nbsp;`string` | API ID of the app on which this event occurred
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`app_api_id` | `null,`&nbsp;`string` | API ID of the app on which this event occurred
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`time` | `int` | Unix timestamp at which the event happened
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`send_id` | `null,`&nbsp;`string` | Message send ID this message belongs to
`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`gender` | `null,`&nbsp;`string` | [PII] gender of the user
`country` | `null,`&nbsp;`string` | [PII] country of the user

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`language` | `null,`&nbsp;`string` | [PII] language of the user
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

### USERS_MESSAGES_WHATSAPP_ABORT_SHARED {#USERS_MESSAGES_WHATSAPP_ABORT_SHARED}

Field | Type | Description
------|------|------------
`id` | `string` | Globally unique ID for this event
`time` | `int` | Unix timestamp at which the event happened
`to_phone_number` | 	`null,`&nbsp;`string` | [PII] phone number of the recipient
`user_id` | `string` | Braze ID of the user that performed this event
`external_user_id` | `null,`&nbsp;`string` | [PII] External user ID of the user
`device_id` | `null,`&nbsp;`string` | `device_id` that is tied to this user if user is anonymous

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`app_group_id` | `null,`&nbsp;`string` | ID of the workspace this user belongs to
`app_group_api_id` | `null,`&nbsp;`string` | API ID of the workspace this user belongs to

> `subscription_group_api_id` | `string` | Subscription group API id

Reviewer: ```suggestion
`subscription_group_api_id` | `string` | Subscription group API ID
```

`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`abort_type` | `null,`&nbsp;`string` | Type of abort, one of: `liquid_abort_message`, `quiet_hours`, `rate_limit`
`abort_log` | `null,`&nbsp;`string` | [PII] log message describing abort details (maximum of 128 characters)
`sf_created_at` | `timestamp`,&nbsp;`null` | When this event was picked up by the Snowpipe      
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

### USERS_MESSAGES_WHATSAPP_DELIVERY_SHARED {#USERS_MESSAGES_WHATSAPP_DELIVERY_SHARED}

Field | Type | Description
------|------|------------
`id` | `string` | Globally unique ID for this event
`time` | `int` | Unix timestamp at which the event happened
`to_phone_number` | `null,`&nbsp;`string` | [PII] phone number of the recipient
`user_id` | `string` | Braze ID of the user that performed this event
`external_user_id` | `null,`&nbsp;`string` | [PII] external user ID of the user
`device_id` | `null,`&nbsp;`string` | `device_id` that is tied to this user if user is anonymous

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`from_phone_number` | `null,`&nbsp;`string` | Phone number from which the WhatsApp message was sent
`app_group_id` | `null,`&nbsp;`string` | ID of the workspace this user belongs to
`app_group_api_id` | `null,`&nbsp;`string` | API ID of the workspace this user belongs to

> `subscription_group_api_id` | `string` | Subscription group API id

Reviewer: ```suggestion
`subscription_group_api_id` | `string` | Subscription group API ID
```

`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`sf_created_at` | `timestamp`,&nbsp;`null` | When this event was picked up by the Snowpipe      
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

### USERS_MESSAGES_WHATSAPP_FAILURE_SHARED {#USERS_MESSAGES_WHATSAPP_FAILURE_SHARED}

Field | Type | Description
------|------|------------
`id` | `string` | Globally unique ID for this event
`time` | `int` | Unix timestamp at which the event happened
`to_phone_number` | `null,`&nbsp;`string` | [PII] phone number of the recipient
`user_id` | `string` | Braze ID of the user that performed this event
`external_user_id` | `null,`&nbsp;`string` | [PII] external user ID of the user
`device_id` | `null,`&nbsp;`string` | `device_id` that is tied to this user if user is anonymous

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`from_phone_number` | `null,`&nbsp;`string` | Phone number from which the WhatsApp message was sent
`app_group_id` | `null,`&nbsp;`string` | ID of the workspace this user belongs to
`app_group_api_id` | `null,`&nbsp;`string` | API ID of the workspace this user belongs to

> `subscription_group_api_id` | `string` | Subscription group API id

Reviewer: ```suggestion
`subscription_group_api_id` | `string` | Subscription group API ID
```

`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`provider_error_code` | `null,`&nbsp;`string` | Error code from WhatsApp
`provider_error_title` | `null, `&nbsp;`string` | Error title from WhatsApp
`sf_created_at` | `timestamp`,&nbsp;`null` | When this event was picked up by the Snowpipe      
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

### USERS_MESSAGES_WHATSAPP_INBOUNDRECEIVE_SHARED {#USERS_MESSAGES_WHATSAPP_INBOUNDRECEIVE_SHARED}

Field | Type | Description
------|------|------------
`id` | `string` | Globally unique ID for this event
`time` | `int` | Unix timestamp at which the event happened
`user_phone_number` | `string` | [PII] the user’s phone number from which the message was received
`user_id` | `string` | Braze ID of the user that performed this event
`external_user_id` | `null,`&nbsp;`string` | [PII] external user ID of the user

> `inbound_phone_number` |

Author: ```suggestion
`inbound_phone_number` | `string` | The inbound number that the message was sent to
```

`device_id` | `null,`&nbsp;`string` | `device_id` that is tied to this user if user is anonymous

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`app_group_id` | `null,`&nbsp;`string` | ID of the workspace this user belongs to
`app_group_api_id` | `null,`&nbsp;`string` | API ID of the workspace this user belongs to

> `subscription_group_api_id` | `string` | Subscription group API id

Reviewer: ```suggestion
`subscription_group_api_id` | `string` | Subscription group API ID
```

`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`message_body` | `string` | Response from the user

> `quick_reply_text` | |

Author: ```suggestion
`quick_reply_text` | `string` | Text of button pressed by the user
```

`media_urls` | `null, {"type"=>"array", "items"=>["null", "string"]}` | Media URLs from the user
`action` | `string` | Action taken in response to this message. For example, `Subscribed`, `Unsubscribed`, or `None`.
`sf_created_at` | `timestamp`,&nbsp;`null` | When this event was picked up by the Snowpipe      
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

### USERS_MESSAGES_WHATSAPP_READ_SHARED {#USERS_MESSAGES_WHATSAPP_READ_SHARED}

Field | Type | Description
------|------|------------
`id` | `string` | Globally unique ID for this event
`time` | `int` | Unix timestamp at which the event happened
`to_phone_number` | `null,`&nbsp;`string` | [PII] phone number of the recipient
`user_id` | `string` | Braze ID of the user that performed this event
`external_user_id` | `null,`&nbsp;`string` | [PII] external user ID of the user
`device_id` | `null,`&nbsp;`string` | `device_id` that is tied to this user if user is anonymous

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`from_phone_number` | `null,`&nbsp;`string` | Phone number from which the WhatsApp message was sent
`app_group_id` | `null,`&nbsp;`string` | ID of the workspace this user belongs to
`app_group_api_id` | `null,`&nbsp;`string` | API ID of the workspace this user belongs to

> `subscription_group_api_id` | `string` | Subscription group API id

Reviewer: ```suggestion
`subscription_group_api_id` | `string` | Subscription group API ID
```

`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to
`sf_created_at` | `timestamp`,&nbsp;`null` | When this event was picked up by the Snowpipe      
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

### USERS_MESSAGES_WHATSAPP_SEND_SHARED {#USERS_MESSAGES_WHATSAPP_SEND_SHARED}

Field | Type | Description
------|------|------------
`id` | `string` | Globally unique ID for this event
`time` | `int` | Unix timestamp at which the event happened
`to_phone_number` | `null,`&nbsp;`string`	| [PII] phone number of the recipient
`user_id` | `string` | Braze ID of the user that performed this event
`external_user_id` | `null,`&nbsp;`string` | [PII] external user ID of the user
`device_id` | `null,`&nbsp;`string` | `device_id` that is tied to this user if user is anonymous

> `timezone` | `null,`&nbsp;`string` | Timezone of the user

Reviewer: ```suggestion
`timezone` | `null,`&nbsp;`string` | Time zone of the user
```

`from_phone_number` | `null,`&nbsp;`string` | phone number from which the WhatsApp message was sent
`app_group_id` | `null,`&nbsp;`string` | ID of the workspace this user belongs to
`app_group_api_id` | `null,`&nbsp;`string` | API ID of the workspace this user belongs to

> `subscription_group_api_id` | `string` | Subscription group API id

Reviewer: ```suggestion
`subscription_group_api_id` | `string` | Subscription group API ID
```

`campaign_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the campaign this event belongs to
`campaign_api_id` | `null,`&nbsp;`string` | API ID of the campaign this event belongs to
`message_variation_api_id` | `null,`&nbsp;`string` | API ID of the message variation this user received
`canvas_id` | `null,`&nbsp;`string` | Internal-use Braze ID of the Canvas this event belongs to
`canvas_api_id` | `null,`&nbsp;`string` | API ID of the Canvas this event belongs to
`canvas_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas variation this event belongs to
`canvas_step_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step this event belongs to
`canvas_step_message_variation_api_id` | `null,`&nbsp;`string` | API ID of the Canvas step message variation this user received
`dispatch_id` | `null,`&nbsp;`string` | ID of the dispatch this message belongs to

> `message_extras` | |

Author: ```suggestion
`message_extras` | `string` | The Liquid tags added to this send event 
```

> `message_extras` | `string` | The Liquid tags added to this send event 

Author: ```suggestion
`message_extras` | `null,`&nbsp;`string` | [PII] A JSON string of the tagged key-value pairs during Liquid rendering
```
